Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/main/MainPresenter.java

```diff
@@ -121,13 +123,31 @@ public void onDirectorySelected(String dir)
 
   public void installWAD(String file)
   {
-    if (NativeLibrary.InstallWAD(file))
-    {
-      Toast.makeText(mContext, R.string.wad_install_success, Toast.LENGTH_SHORT).show();
-    }
-    else
+    final Activity mainPresenterActivity = (Activity) mContext;
+
+    AlertDialog dialog = new AlertDialog.Builder(mContext, R.style.DolphinDialogBase).create();
+    dialog.setTitle("Installing WAD");
+    dialog.setMessage("Installing...");
+    dialog.setCancelable(false);
+    dialog.show();
+
+    Thread installWADThread = new Thread(() ->
     {
-      Toast.makeText(mContext, R.string.wad_install_failure, Toast.LENGTH_SHORT).show();
-    }
+      if (NativeLibrary.InstallWAD(file))
+      {
+        mainPresenterActivity.runOnUiThread(
+                () -> Toast.makeText(mContext, R.string.wad_install_success, Toast.LENGTH_SHORT)
+                        .show());
+      }
+      else
+      {
+        mainPresenterActivity.runOnUiThread(
+                () -> Toast.makeText(mContext, R.string.wad_install_failure, Toast.LENGTH_SHORT)
+                        .show());
+      }
+    }, "InstallWAD");
+    installWADThread.start();
+
+    dialog.dismiss();
```

Comment:
Won't this dismiss the dialog as soon as the thread starts executing, rather than when it's done? It would make more sense to me to do the dismissing from the WAD installation thread.